test/iron-flex-layout.html

```diff
@@ -0,0 +1,412 @@
+      body {
+        margin: 0;
+        padding: 0;
+      }
+      .container {
+        width: 300px;
+        min-height: 50px;
+        background-color: #ccc;
+      }
+      .container > div {
+        width: 50px;
+        min-height: 50px;
+      }
+      /* IE11 does not calculate flex proportions correctly if the children
+       * just have a min-width. For those tests, use width instead */
+      .container > div.fixed-height {
+        min-height: 0;
+        height: 50px;
+      }
+      .relative { @apply(--layout-relative); }
+      .container.relative > div {
+        min-width: 50px;
+        min-height: 50px;
+        width: inherit;
+      }
+      .container.small { width: 120px; }
+      .container.tall { height: 300px; }
+
+      #c1 { background-color: #E91E63; }
+      #c2 { background-color: #03A9F4; }
+      #c3 { background-color: #CDDC39; }
+
+      .horizontal { @apply(--layout-horizontal); }
+      .horizontal-reverse { @apply(--layout-horizontal-reverse); }
+      .vertical { @apply(--layout-vertical); }
+      .vertical-reverse { @apply(--layout-vertical-reverse); }
+      .wrap { @apply(--layout-wrap); }
+      .wrap-reverse { @apply(--layout-wrap-reverse); }
+      .flex { @apply(--layout-flex); }
+      .flex2 { @apply(--layout-flex-2); }
+      .flex3 { @apply(--layout-flex-3); }
+      .center { @apply(--layout-center); }
+      .start { @apply(--layout-start); }
+      .end { @apply(--layout-end); }
+      .start-justified { @apply(--layout-start-justified); }
+      .center-justified { @apply(--layout-center-justified); }
+      .end-justified { @apply(--layout-end-justified); }
+      .justified { @apply(--layout-justified); }
+      .around-justified { @apply(--layout-around-justified); }
+      .fit { @apply(--layout-fit); }
+    </style>
+  </head>
+  <body>
+    <test-fixture id="basic">
+      <template>
+        <div class="container">
+          <div id="c1"></div>
+          <div id="c2"></div>
+          <div id="c3"></div>
+        </div>
+      </template>
+    </test-fixture>
+
+    <test-fixture id="flex">
+      <template>
+        <div class="container">
+          <div id="c1" class="fixed-height"></div>
+          <div id="c2" class="fixed-height"></div>
+          <div id="c3" class="fixed-height"></div>
+        </div>
+      </template>
+    </test-fixture>
+
+    <test-fixture id="single-child">
+      <template>
+        <div class="container">
+          <div id="c1"></div>
+        </div>
+      </template>
+    </test-fixture>
+
+    <test-fixture id="positioning">
+      <template>
+        <div class="container relative">
+          <div id="c1"></div>
+        </div>
+      </template>
+    </test-fixture>
+    <script>
+      function sizeEquals(node, width, height) {
+        var rect = node.getBoundingClientRect();
+        return rect.height === height && rect.width === width;
+      }
+
+      function positionEquals(node, top, bottom, left, right) {
```

Comment:
if anything, it should be in css order (top, right, bottom, left), but I thought top/bottom left/right was easier because it also gives you height/width at a glance